### PR TITLE
feat: add delete account option (attempt 2)

### PR DESCRIPTION
**Pull request by yannickulrich**
This is implementing an account deletion option, similar to #266. I've made the following changes
 * added a `remove_user` function to the database that deletes the user from `users`, `sessions`, and `history`
 * added an API endpoint: currently this is the DELETE handler for the `/register` endpoint but could be moved. If this gets called by a logged in user, the account gets delete
 * added the `unregister` command to the CLI
```shell
$ atuin unregister
```

I'm wondering how we should prevent accidental account deletion. We could ask the user provide their password, email, or username
```shell
$ atuin unregister -e user@example.com
```
Any suggestions? 

PS: Thanks for looking into this and thanks for building this, it's a really cool tool :blush:

### CODE REVIEW

File: atuin-client/src/api_client.rs
Comment (ellie) [inline atuin-client/src/api_client.rs:221]: Could we call this delete please?

File: atuin-common/src/api.rs
Comment (ellie) [inline atuin-common/src/api.rs:23]: Lets also call this one DeleteUserResponse

File: atuin-server/src/database.rs
Comment (ellie) [inline atuin-server/src/database.rs:30]: Again to keep naming consistent, could we call this delete_user please?

File: atuin-server/src/database.rs
Comment (ellie) [inline atuin-server/src/database.rs:341]: same here ;P

File: atuin-server/src/handlers/user.rs
Comment (ellie) [inline atuin-server/src/handlers/user.rs:142]: and here 👀

File: atuin-server/src/handlers/user.rs
Comment (ellie) [inline atuin-server/src/handlers/user.rs:152]:
```suggestion
        return Err(ErrorResponse::reply("failed to delete user")
```

File: atuin/src/command/client/sync/unregister.rs
Comment (ellie) [inline atuin/src/command/client/sync/unregister.rs:1]: If we could also rename this file 🙌

File: docs/docs/commands/sync.md
Comment (ellie) [inline docs/docs/commands/sync.md:35]:
```suggestion
## Delete
```

**Review by ellie — CHANGES_REQUESTED** [review]
Hey! Thank you for your PR! Just a couple of comments

I was actually halfway through implementing this when you opened your PR, so also have a database migration that will make this really fast on the server. I can include my changes once this is in :D

**Comment by yannickulrich** [comment]
Oh wow, that's such a quick response 🤩 I assume you still want to merge this, right? If you've already done a lot, I'm happy to close this..

Assuming you do, I rename it everywhere. Are we keeping the endpoint as it is?

File: atuin-server/src/router.rs
Comment (ellie) [inline atuin-server/src/router.rs:75]:
```suggestion
        .route("/me", delete(handlers::user::unregister))
```

File: atuin-server/src/handlers/user.rs
Comment (yannickulrich) [reply atuin-server/src/handlers/user.rs:152]: oops, sorry, copy & past 😶

**Comment by ellie** [comment]
> Oh wow, that's such a quick response 🤩 I assume you still want to merge this, right? If you've already done a lot, I'm happy to close this..
> 
> Assuming you do, I rename it everywhere. Are we keeping the endpoint as it is?

Hahaha, I was looking at this anyway so all the context is there :D

Yup we can go with this! I'll probs just rebase what I have, as there are a couple of related changes I want to add. Thank you!

File: docs/docs/commands/sync.md
Comment (yannickulrich) [reply docs/docs/commands/sync.md:35]: What do we name the command, still `unregister` or `delete`?

File: atuin-server/src/router.rs
Comment (yannickulrich) [reply atuin-server/src/router.rs:75]: Perfect, I like it 😁

Comment (yannickulrich) [reply atuin-server/src/router.rs]: sorry, I've messed this up, hence the force push

File: atuin-server/src/database.rs
Comment (ellie) [inline atuin-server/src/database.rs:358]: Seeing as we don't need to fetch any data, we can actually just do this instead

```suggestion
    async fn delete_user(&self, u: &User) -> Result<()> {
        sqlx::query::("delete from sessions where user_id = $1")
            .bind(u.id)
            .execute(&self.pool)
            .await?;

        sqlx::query::("delete from users where id = $1")
            .bind(u.id)
            .execute(&self.pool)
            .await?;

        sqlx::query("delete from history where user_id = $1")
            .bind(u.id)
            .execute(&self.pool)
            .await?;

        Ok(())
    }
```

File: docs/docs/commands/sync.md
Comment (ellie) [inline docs/docs/commands/sync.md:40]: I've got some changes I'm making in this area, but I can make them once this is merged 😊

**Review by ellie — COMMENTED** [review]
Just a couple more things then I think we are good to go! 🚀

File: docs/docs/commands/sync.md
Comment (ellie) [reply docs/docs/commands/sync.md]: I'll follow up with a change, I've restructured the commands a bit!

Comment (yannickulrich) [reply docs/docs/commands/sync.md]: Okay cool, in that case I leave it as it is  😊

**Comment by yannickulrich** [comment]
I think I've changed everything you wanted, though I might've missed something somewhere (please let me know if there is)

**Review by ellie — APPROVED** [review]
Great! Thank you for working on this 🙏 

Seeing as this is your first time contributing, if you would like a holographic contributors-only Atuin sticker, then please fill out this [form!](https://notionforms.io/forms/contributors-stickers)

We do also have a [Discord](https://discord.gg/Fq8bJSKPHh) if you'd like to ask any questions, or just fancy hanging out!

Either way - thank you so much for the time and effort here ✨

**Comment by yannickulrich** [comment]
Thanks so much, I really love the idea of the stickers 🤩 Have just filled the form. I still worry about accidental deletion though. Would it not make sense to put some sort of safety in?

**Pull request merged**
